File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        prof = 0
        i = 0
        j = 1
        while j < len(prices):
            if prices[i] < prices[j]:
                prof = max(prof, prices[j] - prices[i])
            else:
                i = j
            j += 1
        return prof

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-5.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> tracker;
        
        int maxFreq = 0;
        int res = 0;
        int i = 0, j = 0;
        while(j < s.size()){
            tracker[s[j]]++;
            maxFreq = max(maxFreq, tracker[s[j]]);

            while(!tracker.empty() && (j - i + 1) - maxFreq > k){
                tracker[s[i++]]--;
            }

            res = max(res, j - i + 1);
            j++;
        }
        return res;
    }
};
